packages/kuma-gui/src/app/zone-egresses/data/index.ts

```diff
@@ -8,39 +8,21 @@ import type {
 
 type PartialInternalZoneEgress = PartialZoneEgressOverview['zoneEgress']
 
-// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
-type InternalZoneEgress = {
-  socketAddress: string
-} & PartialInternalZoneEgress
-
-export type ZoneEgress = {
-  config: PartialZoneEgress
-  socketAddress: string
-} & PartialZoneEgress
-// end TODO
-
 export type ZoneEgressInsight = PartialZoneEgressInsight & DiscoverySubscriptionCollection & {}
 
-// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
-const InternalZoneEgress = {
-  fromObject: (item: PartialInternalZoneEgress): InternalZoneEgress => {
-    return {
-      ...item,
-      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
-    }
-  },
-}
-
 export const ZoneEgress = {
-  fromObject: (item: PartialZoneEgress): ZoneEgress => {
+  fromObject: (item: PartialZoneEgress) => {
     return {
       ...item,
       config: item,
       socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+      networking: {
+        ...item.networking,
+        inboundAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+      },
```

Comment:
I myself have questions over `socketAddress` and `inboundAddress`. Zones already had `socketAddress` but Dataplanes don't. Dataplanes use `networking.inboundAddress` for this feature, so I copied that over here. In a later PR I may change this slightly and try to organize it so it works for all "proxy types" (all proxy types being ingress, egress _and_ dataplane)